api/v1/checks.go

```diff
@@ -220,6 +229,30 @@ func (c JmeterCheck) GetType() string {
 	return "jmeter"
 }
 
+// PopulateConnection will attempt to populate the host and port from the connection name.
+func (c *JmeterCheck) PopulateConnection(ctx context.Context, db *gorm.DB, k8sClient kubernetes.Interface, namespace string) error {
```

Comment:
```suggestion
func (c *JmeterCheck) HydrateConnection(ctx context.Context, db *gorm.DB, k8sClient kubernetes.Interface, namespace string) error {
```

Reply:
I also don't think we need to pass through all the args, we can just pass through context, and then add a context.HydrateConnectionByURL() method that passes through to duty after initializing db / k8s 

If we make this also an interface, we can just call the method on every check if it matches the interface.